rpc/helpers.go

```diff
@@ -38,15 +39,17 @@ func (h *Handler) blockByID(id *BlockID) (*core.Block, *jsonrpc.Error) {
 	case id.Hash != nil:
 		block, err = h.bcReader.BlockByHash(id.Hash)
 	case id.Pending:
-		var pending blockchain.Pending
-		pending, err = h.bcReader.Pending()
-		block = pending.Block
+		var pending *sync.Pending
+		pending, err = h.syncReader.Pending()
+		if err == nil {
+			block = pending.Block
+		}
```

Comment:
Since there is new method `PendingBlock` maybe we should call it here:
```
case id.Pending:
    block = h.syncReader.PendingBlock()
```

Reply:
We could do that but then I will have to add another nil check after the switch statement to check if pending block was found or not. With the current code we get an error, which makes current error handling more consistent.
Please let me know which you would prefer?

Reply:
Agree, let's keep it this way then